### ISSUE TITLE
System.Runtime.Intrinsics.Vector128.Create() bug inside of for loop

### ISSUE THREAD

**Issue by rfellers** [dotnet/runtime#43569]
### Description
I believe that I have found a regression with the way Vector128<float> structs are created using Vector128.Create(float, float, float, float) when converting from doubles. For example, the following code console writes <1, 2, 3, 4> as expected:

```csharp
double[] array = new[] { 1.0, 2, 3, 4 };

Vector128<float> x = Vector128.Create(Convert.ToSingle(array[0]), Convert.ToSingle(array[1]), Convert.ToSingle(array[2]), Convert.ToSingle(array[3]));

Console.WriteLine(x);
```

However, simply putting the code inside a for loop (or while for that matter) causes a console write of <1, 2, 3, ___2___>:

```csharp
double[] array = new[] { 1.0, 2, 3, 4 };

Vector128<float> x = Vector128<float>.Zero;

for (int i = 0; i < 1; i++)
    x = Vector128.Create(Convert.ToSingle(array[0]), Convert.ToSingle(array[1]), Convert.ToSingle(array[2]), Convert.ToSingle(array[3]));

Console.WriteLine(x);
```

### Configuration
dotnet 5.0.100-rc.2.20479.15
Windows 10 Enterprise 2004 (19041.572)
x64 in ___Release___ mode ... in Debug mode, things work as expected.

### Regression
Yes, this is a regression. I tried this using .NET Core 3.1 (3.1.402) and got the expected output either way.

### Other information
One way to work around this issue is to set the 3rd element again manually using WithElement() like so:

```csharp
x = x.WithElement(3, Convert.ToSingle(array[3]));
```

Hope this is helpful, please let me know if you have any questions.

Thanks!

**Comment by EgorBo** [comment]
Codegen bug, in Checked config this method asserts:
```
Assert failure(PID 35100 [0x0000891c], Thread: 18752 [0x4940]): Assertion failed 'tree->gtOper != GT_RET_EXPR' in 'Program:DoWork(double):double' during 'Morph - Inlining' (IL size 54)

    File: flowgraph.cpp Line: 23098
```

**Comment by EgorBo** [comment]
The problematic tree:
```
               [000014] ------------              *  RETURN    double
               [000013] ------------              \--*  CAST      double <- float
               [000012] ------------                 \--*  HWINTRINSIC float  float ToScalar
               [000010] ------------                    \--*  HWINTRINSIC simd16 float Create
               [000009] ------------                       \--*  LIST      void
               [000000] ------------                          +--*  CNS_DBL   float  1.0000000000000000
               [000008] ------------                          \--*  LIST      void
               [000001] ------------                             +--*  CNS_DBL   float  2.0000000000000000
               [000007] ------------                             \--*  LIST      void
               [000002] ------------                                +--*  CNS_DBL   float  3.0000000000000000
               [000006] --C---------                                \--*  LIST      void
               [000005] --C---------                                   \--*  RET_EXPR  float (inl return from call [000015])
```
RET_EXPR was not replaced.

**Comment by EgorBo** [comment]
If I remove this check
![image](https://user-images.githubusercontent.com/523221/96425106-5038c900-1204-11eb-8f49-805559702f24.png)

it starts to work as expected but I guess jit now spends more time. Maybe all GT_HWINTRINSIC nodes should get GTF_CALL on import?

**Comment by AndyAyersMS** [comment]
Reopening to port fix to 5.0.

**Comment by AndyAyersMS** [comment]
Fix is merged into 5.0 and should appear in 5.0.1.
